src/main/java/com/WebController/EventController.java

```diff
@@ -1,66 +1,51 @@
 package com.WebController;
 
-import com.Model.Patient;
-import com.Model.PatientEventAssociation;
-import com.Repository.AssociationRepository;
-import com.Repository.EventRepository;
-import com.Repository.PatientRepository;
-import com.Model.Event;
+import com.Model.*;
+import com.Repository.*;
+import com.Service.*;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Controller;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
-
-import java.util.ArrayList;
 import java.util.List;
 
-@Controller
+@RestController
 public class EventController {
 
-    @Autowired
-    private PatientRepository patientRepo;
-
     @Autowired
     private EventRepository eventRepo;
 
     @Autowired
-    private AssociationRepository assoicationRepo;
-
-    @GetMapping("/")
-    public String homePage() {
-        return "index";
-    }
-
+    private EventService eventService;
+
+    /**
+     * Handler for patients creation.
+     * User create a patient in front end and associate patient with multiple events.
+     * Data posted with JSON string and parsed to Event objects directly.
+     * Event duplication should be avoided.
+     */
     @PostMapping(value = "/patientinfo", consumes = "application/json")
-    @ResponseBody
-    public String receivePatient (@RequestBody List<Event> events){
-        Patient tempPatient = new Patient();
-        List<Event> eventList = new ArrayList<>();
-        Iterable<Event> allEvents = this.eventRepo.findAll();
-        allEvents.forEach(eventList::add);
-        PatientEventAssociation association;
-        for (Event e: events){
-            int index = eventList.indexOf(e);
-            if (index<0) {//new Event, save it to Repo
-                association = new PatientEventAssociation(e, tempPatient, true);
-                e.addAssociation(association);
-                eventRepo.save(e);
-            } else {//event already exist, just save association
-                Event event = eventList.get(index);
-                association = new PatientEventAssociation(event, tempPatient, true);
-                event.addAssociation(association);
-                eventRepo.save(event);
-            }
+    public ResponseEntity<String> receivePatient (@RequestBody List<Event> events){
+        try {
+            this.eventService.saveEvents(events);
```

Comment:
See, if could just return the result of saving the events in the event repo, you will be calling a CrudRepo that already is a REST endpoint, and it will take care of the proper HTTP codes and behavior. Let Spring do the hard work for you. Minimize creating needless controllers!

I'd go back to the way you had this before, and just `return eventRepo.save(event)` when applicable.

Reply:
The return type of eventRepo.save(event) is Event, since user may save a list of events, I will return a list of Events as well.

Reply:
Maybe it's just better to add one event at a time, and change the API accordingly. Once sessions are in place it will be possible to track who added the event, and determine if it's a verified event or not.

Reply:
If so, we may need to delete the entire Patient class, because if we just receive one event a time, system won't be able to tell the event belongs to which specific patient. Are we going to do this way?

Reply:
How will you deal with adding more events for the same anonymous user later on? Or should we just consider that a new/different user?

Reply:
For now, we create a new user every time when users create new events.

Reply:
I think a solution would be to generate a session id (but not a cookie) when a new patient is created, and associate the patient with that session. When a new event is added, you retrieve the session and add the event to that patient.